apis/sfn/v1alpha1/zz_state_machine.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// StateMachineParameters defines the desired state of StateMachine
+type StateMachineParameters struct {
+	// Region is which region the StateMachine will be created.
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+	// The Amazon States Language definition of the state machine. See Amazon States
+	// Language (https://docs.aws.amazon.com/step-functions/latest/dg/concepts-amazon-states-language.html).
+	// +kubebuilder:validation:Required
+	Definition *string `json:"definition"`
```

Comment:
Should this be a *string or a string?

Reply:
Yeah, it should be a `string` but ACK copies the types from AWS Go SDK directly so we get everything as pointers.